src/api/dataops/dataOperationsModule.ts

```diff
@@ -5,6 +5,16 @@ import { AuthOptions } from "../core/tokenManager";
 import { DataSetName } from "./dataops.tokens";
 import { Dataset } from "./dataset";
 
+/**
+ * @internal
+ */
+function datasetNameErrorFactory() {
+  // tslint:disable-next-line:only-arrow-functions
+  return function() {
+    throw new Error("Please set the dataset name at the DI");
```

Comment:
What is the scenario which this can happen for the final user? Is it if they call `dataOperations.dataset(null)`?
If that's the case maybe we could make this message more friendly like

```suggestion
    throw new Error('Datasets requires valid names, please assure you provide a string value. \ne.g. dataOperations.dataset('authors')');
```

Reply:
Changed 👍